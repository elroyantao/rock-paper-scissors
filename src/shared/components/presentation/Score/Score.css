.Score {
  display: flex;
  justify-content: space-between;
}

.Score-display {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
}

.Score-label {
  font-size: 18px;
}

.Score-current {
  font-size: 15px;
}
