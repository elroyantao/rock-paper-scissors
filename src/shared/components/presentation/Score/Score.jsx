
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import styles from './Score.css'

@connect((state) => ({
  playerScore: state.score.player,
  computerScore: state.score.computer
}), {})
export default class Score extends Component {

  static propTypes = {
    playerScore: PropTypes.number,
    computerScore: PropTypes.number
  }

  render() {
    const { playerScore, computerScore } = this.props
    return (
      <section className={styles.Score}>
        <div className={styles['Score-display']}>
          <div className={styles['Score-label']}>
            Player
          </div>
          <div className={styles['Score-current']}>
            {playerScore}
          </div>
        </div>
        <div className={styles['Score-display']}>
          <div className={styles['Score-label']}>
            Computer
          </div>
          <div className={styles['Score-current']}>
            {computerScore}
          </div>
        </div>
      </section>
    )
  }
}
