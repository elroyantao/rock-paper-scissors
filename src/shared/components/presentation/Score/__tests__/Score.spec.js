import testComponentHelper from '../../../../lib/test-helper'
import Score from '../Score'


describe('<Score /> ', () => {
  const initialState = {
    playerScore: 0,
    computerScore: 0
  }
  const renderComponent = testComponentHelper(Score.WrappedComponent)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialState).getTree())
        .toMatchSnapshot()
    })
    it('when score is player 6 - 4 computer', () => {
      expect(renderComponent({
        playerScore: 6,
        computerScore: 4
      }).getTree())
        .toMatchSnapshot()
    })
  })
})
