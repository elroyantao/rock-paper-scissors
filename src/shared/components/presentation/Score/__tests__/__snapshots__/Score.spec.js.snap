// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score />  @renders in default state 1`] = `
<section
  className="Score"
>
  <div
    className="Score-display"
  >
    <div
      className="Score-label"
    >
      Player
    </div>
    <div
      className="Score-current"
    >
      0
    </div>
  </div>
  <div
    className="Score-display"
  >
    <div
      className="Score-label"
    >
      Computer
    </div>
    <div
      className="Score-current"
    >
      0
    </div>
  </div>
</section>
`;

exports[`<Score />  @renders when score is player 6 - 4 computer 1`] = `
<section
  className="Score"
>
  <div
    className="Score-display"
  >
    <div
      className="Score-label"
    >
      Player
    </div>
    <div
      className="Score-current"
    >
      6
    </div>
  </div>
  <div
    className="Score-display"
  >
    <div
      className="Score-label"
    >
      Computer
    </div>
    <div
      className="Score-current"
    >
      4
    </div>
  </div>
</section>
`;
